### PR TITLE
Weapon swap rework, fire on round end

### PR DESCRIPTION
**Pull request by matekdev**

### CODE REVIEW

File: code/Player/Inventory.cs [hunk 1]
```diff
@@ -11,6 +11,27 @@ public partial class Inventory : EntityComponent<Player>
 	[Net, Predicted]
 	public Weapon LastActiveWeapon { get; private set; }
 
+	private Weapon _lastSwappedWeapon { get; set; }
```
Comment (matekdev) [inline code/Player/Inventory.cs:14]: Got a better name?

File: code/Weapons/Weapon.cs [hunk 2]
```diff
@@ -99,11 +98,12 @@ public void OnDeploy( Grub grub )
...
 
 		if ( FiringType is FiringType.Cursor )
-			SetPointerEvents( To.Single( Grub ), true );
+			Event.Run( GrubsEvent.Player.PointerEventChanged, true );
 	}
 
-	public void Holster()
+	public void Holster( Grub _ )
 	{
+		SetParent( null );
```
Comment (matekdev) [inline code/Weapons/Weapon.cs:106]: Not sure if this is needed / we should include this

File: code/Events/Event.Grub.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+public static partial class GrubsEvent
+{
+	public static class Grub
+	{
+		public const string Damaged = "grub.damaged";
+
+		/// <summary>
+		/// Occurs when a grub is damaged.
+		/// <para><see cref="int"/> The network ident of the damaged grub</para>
```
Comment (trundler-dev) [inline code/Events/Event.Grub.cs:11]: Just fix up the spacing here:
```suggestion
		/// <para><see cref="int"/>The network ident of the damaged grub</para>
```

File: code/Events/Event.Player.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+public static partial class GrubsEvent
+{
+	public static class Player
+	{
+		public const string PointerEventChanged = "player.pointer.event";
+
+		/// <summary>
+		/// Occurs if the pointer event changes.
+		/// <para><see cref="bool"/> If the pointer events are enabled </para>
```
Comment (trundler-dev) [inline code/Events/Event.Player.cs:11]:
```suggestion
		/// <para><see cref="bool"/>If the pointer events are enabled</para>
```

File: code/Player/Inventory.cs (hunk 1 above)
Comment (trundler-dev) [reply code/Player/Inventory.cs:14]: Nope, that's good 👍

File: code/Weapons/Weapon.cs (hunk 2 above)
Comment (trundler-dev) [reply code/Weapons/Weapon.cs:106]: I think it's redundant to include. When we parent on deploy it's just to set it up for the first time - it should stay parented no matter what.

**Review by trundler-dev — APPROVED** [review]

**Pull request merged**